odyssey/client/src/Profiles/UserProfile.svelte

```diff
@@ -74,6 +85,56 @@
 
 	}
 
+	async function isLiked(post_id) {
+		const response = await fetchPost('http://localhost:3000/hasLikedPost', {
+			post_id: post_id
+		});
+
+		let temp = response.liked;
+
+		return temp;
+	}
+
+	async function canView(post_id) {
+		const response = await fetchPost('http://localhost:3000/canViewPost', {
+			post_id: post_id
+		});
+
+		let temp = response.view;
+
+		return temp;
+	}
+
+	async function likePost(post_id) {
+		const response = await fetchPost('http://localhost:3000/likePost', {
+			post_id: post_id
+		});
+
+		posts.find(post => post._id.$oid === post_id).isLiked = true;
+		posts = posts;
```

Comment:
Hi! Can you explain why is this needed? I see everything is working just fine, but I'd really appreciate if you could explain this line to me. Thanks! 😄

Reply:
line 114

Reply:
You can check out [this link](https://svelte.dev/tutorial/updating-arrays-and-objects) from Svelte Docs. Basically when you set `post.isLiked = true;` this won't update references to `posts`, which is why it should be followed by `posts = posts`.

_A simple rule of thumb: the name of the updated variable must appear on the left hand side of the assignment._ - Svelte Docs

Reply:
Thanks! ❤️